Documentation/Functions/Data.rst

```diff
@@ -46,726 +45,815 @@ Data / getText
+..  note::
+    It is safe to directly use a client-/user-provided input for the id of a DB
+    record here. The function :php:`ContentObjectRenderer->getData` internally
+    calls the function :php:`PageRepository->getRawRecord`, which converts the
+    parameter to int via :php:`QueryBuilder->createNamedParameter`
 
-.. _data-type-gettext-debug:
+..  _data-type-gettext-debug:
 
 debug
 -----
 
-:aspect:`getText Key:`
-   debug
 
-:aspect:`Description:`
-   Returns HTML-formatted content of the PHP variable defined by the keyword.
-   Available keywords are :typoscript:`rootLine`, :typoscript:`fullRootLine`, :typoscript:`data`,
-   :typoscript:`register` and :typoscript:`page`.
+..  confval:: debug
+    :name: data-debug
+
+    Returns HTML-formatted content of the PHP variable defined by the keyword.
+    Available keywords are :typoscript:`rootLine`, :typoscript:`fullRootLine`, :typoscript:`data`,
+    :typoscript:`register` and :typoscript:`page`.
+
+..  _data-type-gettext-debug-example:
+
+Example: Debug the current root-line
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-:aspect:`Example:`
-   Outputs the current root-line visually in HTML:
 
-   .. code-block:: typoscript
-      :caption: EXT:site_package/Configuration/TypoScript/setup.typoscript
+Outputs the current root-line visually in HTML:
 
-      lib.foo.data = debug : rootLine
+..  code-block:: typoscript
+    :caption: EXT:site_package/Configuration/TypoScript/setup.typoscript
 
-.. _data-type-gettext-field:
+    lib.foo.data = debug : rootLine
+
+..  _data-type-gettext-field:
 
 field
 -----
 
-:aspect:`getText Key:`
-   field
+..  confval:: field
+    :name: data-field
+    :Syntax: field : [field name from the current :php:`$cObj->data` array in the cObject, multi-dimensional.]
+
+
+    This gives read access to the current value of an internal global variable determined by the given key.
+
+    -   As default the :php:`$cObj->data` array is :php:`$GLOBALS['TSFE']->page`
+        (record of the current page)
+
+    -   In :ref:`TMENU <tmenu>` :php:`$cObj->data` is set in a loop to the page-record for
+        each menu item during its rendering process.
 
-:aspect:`Syntax:`
-   field : [field name from the current :php:`$cObj->data` array in the cObject, multi-dimensional.]
+    -   In :ref:`cobj-content` / :ref:`cobj-records` :php:`$cObj->data` is set to
+        the actual record
 
-:aspect:`Description:`
-   This gives read access to the current value of an internal global variable determined by the given key.
+    -   In :ref:`GIFBUILDER <gifbuilder>` :php:`$cObj->data` is set to the data
+        :ref:`GIFBUILDER <gifbuilder>` is supplied with.
 
-   - As default the :php:`$cObj->data` array is :php:`$GLOBALS['TSFE']->page`
-     (record of the current page)
+..  _data-type-gettext-field-example:
 
-   - In :ref:`TMENU <tmenu>` :php:`$cObj->data` is set in a loop to the page-record for
-     each menu item during its rendering process.
+Examples
```

Comment:
Do we really need this header here? The anchor above can also point to the first example. This way, we don't have a 5th level of headings.